metricbeat/beater/metricbeat.go

```diff
@@ -68,6 +86,12 @@ func (bt *Metricbeat) Run(b *beat.Beat) error {
 		PublishChannels(bt.client, cs...)
 	}()
 
+	// Blocks metricbeat from shutting down if no module is set. Required for config reloading
+	if bt.config.ReloadModules.Enabled {
```

Comment:
I think the initialization and starting of the NewConfigReloader should happen here to so that it's all in one place. You could make everything block on the same WaitGroup too...

```
if bt.config.ReloadModules.Enabled {
  configReloader := NewConfigReloader..

  // Start
  wg.Add(1)
  go func() {
    defer wg.Done()
    configReloader.Run()
  }()

  // Stop
  go func() {
    <-bt.done
    configReloader.Stop()
  }()
}